src/Checkout/ExpressCheckout/Service/ExpressCheckoutDataServiceInterface.php

```diff
@@ -7,12 +7,17 @@
 
 namespace Swag\PayPal\Checkout\ExpressCheckout\Service;
 
+use Shopware\Core\Framework\Event\ShopwareSalesChannelEvent;
 use Shopware\Core\Framework\Log\Package;
 use Shopware\Core\System\SalesChannel\SalesChannelContext;
 use Swag\PayPal\Checkout\ExpressCheckout\ExpressCheckoutButtonData;
 
 #[Package('checkout')]
 interface ExpressCheckoutDataServiceInterface
 {
-    public function buildExpressCheckoutButtonData(SalesChannelContext $salesChannelContext, bool $addProductToCart = false): ?ExpressCheckoutButtonData;
+    public function buildExpressCheckoutButtonData(
+        SalesChannelContext $salesChannelContext,
+        bool $addProductToCart = false,
+        ?ShopwareSalesChannelEvent $event = null
+    ): ?ExpressCheckoutButtonData;
```

Comment:
the reason, why this interface exists, is, because the class is overwritten in Rufus. Please check there to make the appropriate adjustments (and backport to 6.7.0.0 there)

Reply:
I can not find this Interface being used in [Rufus](https://github.com/search?q=repo%3Ashopware%2FRufus+ExpressCheckoutDataServiceInterface&type=code).

Reply:
How about we kill it then, if we break it anyway?

Reply:
Hmmmm isn't removing of the Interface a breaking change anyway? So I can deprecate it, but I do think having an Interface is better

Reply:
Both removing and changing parameters to an interface is a breaking change, so no difference there.

I don't see the benefit of this singular interface then, since all other Storefront Data Providers don't have interfaces, that's why I thought we could just remove it. AFAIK it is there, because in previous versions, we needed to overwrite stuff in Rufus, but don't anymore.

@cyl3x third opinion?